src/odemis/acq/align/test/autofocus_test.py

```diff
@@ -602,5 +603,76 @@ def test_multi_det(self):
         self.assertEqual(len(res), ngs * nds)
 
 
+class TestAutofocus1d(unittest.TestCase):
+    """
+    Test autofocus functions on 1 line CCD.
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            test.start_backend(SPARC2_FOCUS_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # find components by their role
+        cls.ccd = model.getComponent(role="ccd")
+        cls.spectrometer = model.getComponent(role="spectrometer-integrated")
+
+        cls.focus = model.getComponent(role="focus")
+        cls._good_focus = cls.focus.position.value["z"]
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    @timeout(1000)
+    def test_autofocus_spect(self):
+        """
+        Test AutoFocus on 1 line CCD for example spectrum.
+        """
+        self.focus.moveAbs({"z": self._good_focus - 200e-6}).result()
+        self.ccd.updateMetadata({model.MD_LENS_RI: 1.0e-2})
```

Comment:
If they all need this, I'd recommend to move this to the `setUpClass()`, or even in the microscope file. 
BTW, what is this for? IIRC, that's for depthOfField, which influences the minimum step? A RI of 0.01 seems really weird. I wonder if instead we could just increase the NA, or maybe increase the focus step size...